HW03/vorobyov_egor/spec/logs_parser_spec.rb

```diff
@@ -0,0 +1,124 @@
+# frozen_string_literal: true
+
+require '../lib/logs_parser'
+
+RSpec.describe LogsParser do
+  let(:logs) do
+    <<~LOGS
+      10.6.246.103 - - [23/Apr/2018:20:30:39 +0300] 'POST /test/2/messages HTTP/1.1' 200 48 0.0498
+      10.6.246.101 - - [23/Apr/2018:20:30:42 +0300] 'POST /test/2/run HTTP/1.1' 200 - 0.2277
+      2018-04-23 20:30:42: SSL ERROR, peer: 10.6.246.101, peer cert: , #<Puma::MiniSSL::SSL: System error: Undefined error: 0 - 0>
+      10.6.246.101 - - [23/Apr/2018:20:31:39 +0300] 'POST /test/2/messages HTTP/1.1' 200 48 0.0290
+    LOGS
+  end
+
+  let(:time_arr) do
+    ['2018-04-23 17:17:49.7',
+     '2018-04-23 17:18:38.8',
+     '2018-04-23 17:18:39.8']
+  end
+
+  let(:time_interval) { [49.1, 1.0] }
+
+  describe '.task_1' do
+    context 'when arr of logs was passed' do
+      let(:error) do
+        '2018-04-23 20:30:42: SSL ERROR, peer: 10.6.246.101, peer cert: , #<Puma::MiniSSL::SSL: System error: Undefined error: 0 - 0>'
+      end
+
+      it 'returns the first string with error' do
+        expect(LogsParser.task_1(logs).chomp).to eq error
+      end
+    end
+
+    context 'when arr wasn\'n passed' do
+      it 'returns ArgumentError' do
+        expect { LogsParser.task_1 }.to raise_error(ArgumentError)
+      end
+    end
+
+    context 'when arr wasn\'n passed' do
+      it 'returns NoMethodError' do
+        expect { LogsParser.task_1([1, true, 'str', 4, 5]) }.to raise_error(NoMethodError)
+      end
+    end
+  end
+
+  describe '.task_2' do
+    context 'when logs were passed' do
+      let(:formatted) do
```

Comment:
We should use a more clear name
```suggestion
      let(:formatted_logs) do
```